src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -546,19 +576,22 @@ namespace nbl::ext::imgui
 		//}
 	//}
 
-	bool UI::Render(IGPUCommandBuffer* commandBuffer, int const frameIndex)
+	bool UI::render(IGPUCommandBuffer* commandBuffer, const uint32_t frameIndex)
 	{
+		const bool validFramesRange = frameIndex >= 0 && frameIndex < maxFramesInFlight;
```

Comment:
btw also take `SIntededSubmit` instead of just `commandBuffer` then you'll be able to handle overflows nicely and also steal the `signalSemaphores` to latch the free of the mdiBuffer allocation upon a frame completion

Reply:
resolved

Reply:
we agreed [we remove](https://github.com/Devsh-Graphics-Programming/Nabla/commit/18949b17da5c4cab8386975afbf5a24a9c9ee36f) `SIntededSubmit` from the render method due to overflow & non-dynamic renderpass issues